### PR TITLE
gh-119146: Update ``regexp`` in `build.yml` to not trigger the jobs on `*.md` and `*.ini` files.

### PR DESCRIPTION
**Pull request by Eclips4**
* Issue: gh-119146

### CODE REVIEW

**Comment by Eclips4** [comment]
@hugovk 
So, these wildcard patterns actually work and the jit workflow is skipped, but other workflows don't take these exceptions into account.

**Comment by Eclips4** [comment]
I've misinterpreted the run of the builds, because changing `build.yml` is triggers them and I thought this was wrong... Sorry for the mess.

Actually, we just need to add `*.md` and `*.ini` files to our regexp to make things work.

**Review by savannahostrowski — APPROVED** [review]
The commit messages are so relatable 😅 . Testing GHA workflows is always such a pain. This LGTM!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
             # into the PR branch anyway.
             #
             # https://github.com/python/core-workflow/issues/373
-            git diff --name-only origin/$GITHUB_BASE_REF.. | grep -qvE '(\.rst$|^Doc|^Misc|^\.pre-commit-config\.yaml$|\.ruff\.toml$)' && echo "run_tests=true" >> $GITHUB_OUTPUT || true
+            git diff --name-only origin/$GITHUB_BASE_REF.. | grep -qvE '(\.rst$|^Doc|^Misc|^\.pre-commit-config\.yaml$|\.ruff\.toml$|\.md$|\.ini$)' && echo "run_tests=true" >> $GITHUB_OUTPUT || true
```
Comment (hugovk) [inline .github/workflows/build.yml]: Most of the `.ini` files are `mypy.ini` that we can skip.

But one is `Tools/freeze/extensions_win32.ini`, should that trigger a build?

Comment (Eclips4) [reply .github/workflows/build.yml]: I guess yes, it should trigger a Windows build at least.
What is the preferable way to exclude this .ini file? Modify the regexp?

Comment (hugovk) [reply .github/workflows/build.yml]: Yeah, exclude `mypy\.ini$`, something like this?
```suggestion
            git diff --name-only origin/$GITHUB_BASE_REF.. | grep -qvE '(\.rst$|^Doc|^Misc|^\.pre-commit-config\.yaml$|\.ruff\.toml$|\.md$|mypy\.ini$)' && echo "run_tests=true" >> $GITHUB_OUTPUT || true
```

**Review by hugovk — APPROVED** [review]
Thanks! Shall we also backport to 3.12?

**Comment by Eclips4** [comment]
> Thanks! Shall we also backport to 3.12?

I think yes. https://github.com/python/cpython/pull/119147 wasn't backported to 3.12 because it was related to JIT workflow which appeared only in 3.13.
This PR does a more generally thing, so I think it would be ok to backport it to 3.12.

**Pull request merged**